functions/Remove-DbaDbMailProfile.ps1

```diff
@@ -0,0 +1,125 @@
+
+        dbatools PowerShell module (https://dbatools.io)
+        Copyright: (c) 2021 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Remove-DbaDbMailProfile
+
+    .EXAMPLE
+        PS C:\> Remove-DbaDbMailProfile -SqlInstance localhost, localhost\namedinstance
+
+        Removes all database mail profiles on the localhost, localhost\namedinstance instances.
+
+    .EXAMPLE
+        PS C:\> Remove-DbaDbMailProfile -SqlInstance localhost -Name MyDatabaseMailProfile
+
+        Removes MyDatabaseMailProfile database mail profile on the localhost.
+
+    .EXAMPLE
+        PS C:\> Get-DbaDbMailProfile -SqlInstance SRV1 | Out-GridView -Title 'Select database mail profile(s) to drop' -OutputMode Multiple | Remove-DbaDbMailProfile
+
+        Using a pipeline this command gets all database mail profiles on SRV1, lets the user select those to remove and then removes the selected database mail profiles.
+    #>
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'High')]
+    param (
+        [Parameter(ParameterSetName = 'NonPipeline', Mandatory = $true, Position = 0)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [Parameter(ParameterSetName = 'NonPipeline')]
+        [PSCredential]$SqlCredential,
+        [Parameter(ParameterSetName = 'NonPipeline')]
+        [string[]]$Name,
+        [Parameter(ParameterSetName = 'NonPipeline')]
+        [string[]]$ExcludeProfile,
+        [parameter(ValueFromPipeline, ParameterSetName = 'Pipeline', Mandatory = $true)]
+        [Microsoft.SqlServer.Management.Smo.Mail.MailProfile[]]$InputObject,
+        [Parameter(ParameterSetName = 'NonPipeline')][Parameter(ParameterSetName = 'Pipeline')]
+        [switch]$EnableException
+    )
+
+    begin {
+        $dbMailProfiles = @( )
+    }
+
+    process {
+        if ($SqlInstance) {
+            $params = $PSBoundParameters
+            $null = $params.Remove('WhatIf')
+            $null = $params.Remove('Confirm')
+            $dbMailProfiles = Get-DbaDbMailProfile @params
+        } else {
+            $dbMailProfiles += $InputObject
+        }
+    }
+
+    end {
+        # We have to delete in the end block to prevent "Collection was modified; enumeration operation may not execute." if directly piped from Get-DbaDbMailProfile.
+        foreach ($dbMailProfile in $dbMailProfiles) {
+            if ($PSCmdlet.ShouldProcess($dbMailProfile.Parent.Parent.Name, "Removing the database mail profile $($dbMailProfile.Name) in the database $($dbMailProfile.Parent.Name) on $($dbMailProfile.Parent.Parent.Name)")) {
+                $output = [pscustomobject]@{
+                    ComputerName  = $dbMailProfile.Parent.Parent.ComputerName
+                    InstanceName  = $dbMailProfile.Parent.Parent.ServiceName
+                    SqlInstance   = $dbMailProfile.Parent.Parent.DomainInstanceName
+                    DbMailProfile = $dbMailProfile.Name
```

Comment:
```suggestion
                    Name = $dbMailProfile.Name
```

Matches output for the same property as the Get command.

Reply:
Will that be consistent with the proposed parameter change : Name -> Profile?

Reply:
Done. Changed to match outputs of the Get / New commands.